pkg/scheduler/factory/plugins.go

```diff
@@ -385,7 +385,7 @@ func RegisterCustomPriorityFunction(policy schedulerapi.PriorityPolicy) string {
 						policy.Argument.LabelPreference.Presence,
 					)
 				},
-				Weight: policy.Weight,
+				Weight: int(policy.Weight),
```

Comment:
> unless there's a strong reason not to, keeping int64 for compatibility with all possible existing config files is preferable

@ahg-g  @liggitt @alculquicondor  I think this would be a reason to have `PriorityPolicy.Weight` as int32 and not int64, otherwise we would change the type of `PriorityConfigFactory.Weight` to int64. What do you think?